cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -378,6 +431,10 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) errors.AutoscalerError
 	}
 
 	if a.ScaleDownEnabled {
+
+		// TODO(scheduler_framework migration) should we repropagate ClusterSnaphshot before scaled down?
+		// If we follow old logic it seems we should not.
```

Comment:
I think we should at minimum filter out upcoming nodes. We should not delete node because its pods can go to upcoming node which may or may not actually show up. Probably the right way to filter them out would be to remove them from podDestinations list and not snapshot though.

Reply:
Actually we are filtering those out. Kinda. As we are building `podDestinations` based on `allNodes`. And `allNodes` are nodes actually read from k8s. Not those from `ClusterSnapshot`. Maybe we should clean that up?

Reply:
I'll add it to follow-up list.